src/modules/profile/pages/profileSettings/components/CreatorNotifications.tsx

```diff
@@ -0,0 +1,270 @@
+
+  const getConfigValue = (
+    creatorNotificationSettings: CreatorNotificationSettings,
+    type: NotificationType,
+    name: ConfigName,
+  ) => {
+    const setting = creatorNotificationSettings.notificationSettings.find((s) => s.notificationType === type)
+    return setting?.configurations.find((c) => c.name === name)?.value
+  }
+
+  const getConfigId = (
+    creatorNotificationSettings: CreatorNotificationSettings,
+    type: NotificationType,
+    name: ConfigName,
+  ) => {
+    const setting = creatorNotificationSettings.notificationSettings.find((s) => s.notificationType === type)
+    return setting?.configurations.find((c) => c.name === name)?.id
+  }
+
+  const updateSettingsMap = (
+    prevSettings: CreatorNotificationSettings[],
+    projectId: string,
+    type: NotificationType,
+    name: ConfigName,
+    newValue: string,
+  ) => {
+    return prevSettings.map((setting) => {
+      if (setting.project.id === projectId) {
+        return {
+          ...setting,
+          notificationSettings: setting.notificationSettings.map((notifSetting) => {
+            if (notifSetting.notificationType === type) {
+              return {
+                ...notifSetting,
+                configurations: notifSetting.configurations.map((config) =>
+                  config.name === name ? { ...config, value: newValue } : config,
+                ),
+              }
+            }
+
+            return notifSetting
+          }),
+        }
+      }
+
+      return setting
+    })
+  }
+
+  const updateConfigValue = async (
+    creatorNotificationSettings: CreatorNotificationSettings,
+    type: NotificationType,
+    name: ConfigName,
+    value: string,
+  ) => {
+    const configId = getConfigId(creatorNotificationSettings, type, name)
+    if (!configId) return
+
+    // Optimistically update the UI
+    setSettings((prevSettings) =>
+      updateSettingsMap(prevSettings, creatorNotificationSettings.project.id, type, name, value),
+    )
+
+    try {
+      await updateNotificationSetting({
+        variables: {
+          creatorNotificationConfigurationId: configId,
+          value,
+        },
+      })
+
+      // Show success toast
+      toast({
+        title: 'Update successful',
+        description: 'Notification setting has been updated.',
+        status: 'success',
+        duration: 3000,
+        position: 'top-right',
+      })
+    } catch (error) {
+      // Revert the optimistic update
+      setSettings((prevSettings) =>
+        updateSettingsMap(
+          prevSettings,
+          creatorNotificationSettings.project.id,
+          type,
+          name,
+          getConfigValue(creatorNotificationSettings, type, name) || '',
+        ),
+      )
+
+      // Show error toast
+      toast({
+        title: 'Update failed',
+        description: 'Failed to update notification setting. Please try again.',
+        status: 'error',
+        duration: 5000,
+        position: 'top-right',
+      })
```

Comment:
Using the above mentioned hook this will be reduced to:

```suggestion
      toast.error({
        title: 'Update failed',
        description: 'Failed to update notification setting. Please try again.',
      })
```